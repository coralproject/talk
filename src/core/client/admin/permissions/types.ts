import { GQLUSER_ROLE_RL } from "coral-framework/schema";

export type PermissionMap<AbilityType extends string, ContextType> = {
  [abilityType in AbilityType]: {
    [role in GQLUSER_ROLE_RL]?: (context?: ContextType) => boolean;
  };
};
