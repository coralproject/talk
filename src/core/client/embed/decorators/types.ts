import pym from "pym.js";

export type CleanupCallback = () => void;
export type Decorator = (pym: pym.Parent) => CleanupCallback | void;
