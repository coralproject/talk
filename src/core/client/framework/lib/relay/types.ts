import { _RefType } from "react-relay";

export type FragmentKeys<T> = {
  [P in keyof T]: T[P] extends _RefType<any> | null ? P : never
}[keyof T];

export type FragmentKeysNoLocal<T> = Exclude<FragmentKeys<T>, "local">;
