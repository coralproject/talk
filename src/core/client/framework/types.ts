// TODO: (@cvle) Extract useful common types into its own package.
export { Diff, Omit, Overwrite, PropTypesOf } from "talk-ui/types";
