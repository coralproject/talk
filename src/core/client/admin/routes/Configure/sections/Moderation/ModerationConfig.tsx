import React, { FunctionComponent } from "react";

import { PropTypesOf } from "coral-framework/types";
import { HorizontalGutter } from "coral-ui/components";

import AkismetConfigContainer from "./AkismetConfigContainer";
import PerspectiveConfigContainer from "./PerspectiveConfigContainer";
import PreModerationConfigContainer from "./PreModerationConfigContainer";
import RecentCommentHistoryConfigContainer from "./RecentCommentHistoryConfigContainer";

interface Props {
  disabled: boolean;
  settings: PropTypesOf<typeof AkismetConfigContainer>["settings"] &
    PropTypesOf<typeof PerspectiveConfigContainer>["settings"] &
    PropTypesOf<typeof PreModerationConfigContainer>["settings"] &
    PropTypesOf<typeof RecentCommentHistoryConfigContainer>["settings"];
  onInitValues: (values: any) => void;
}

const ModerationConfig: FunctionComponent<Props> = ({
  disabled,
  settings,
  onInitValues,
}) => (
  <HorizontalGutter size="double" data-testid="configure-moderationContainer">
    <PreModerationConfigContainer
      disabled={disabled}
      settings={settings}
      onInitValues={onInitValues}
    />
    <RecentCommentHistoryConfigContainer
      disabled={disabled}
      settings={settings}
      onInitValues={onInitValues}
    />
    <PerspectiveConfigContainer
      disabled={disabled}
      settings={settings}
      onInitValues={onInitValues}
    />
    <AkismetConfigContainer
      disabled={disabled}
      settings={settings}
      onInitValues={onInitValues}
    />
  </HorizontalGutter>
);

export default ModerationConfig;
