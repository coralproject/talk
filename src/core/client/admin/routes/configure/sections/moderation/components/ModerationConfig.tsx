import React, { StatelessComponent } from "react";

import { PropTypesOf } from "talk-framework/types";
import { HorizontalGutter } from "talk-ui/components";

import AkismetConfigContainer from "../containers/AkismetConfigContainer";
import PerspectiveConfigContainer from "../containers/PerspectiveConfigContainer";

interface Props {
  disabled: boolean;
  settings: PropTypesOf<typeof AkismetConfigContainer>["settings"] &
    PropTypesOf<typeof PerspectiveConfigContainer>["settings"];
  onInitValues: (values: any) => void;
}

const ModerationConfig: StatelessComponent<Props> = ({
  disabled,
  settings,
  onInitValues,
}) => (
  <HorizontalGutter size="double" data-testid="configure-moderationContainer">
    <PerspectiveConfigContainer
      disabled={disabled}
      settings={settings}
      onInitValues={onInitValues}
    />
    <AkismetConfigContainer
      disabled={disabled}
      settings={settings}
      onInitValues={onInitValues}
    />
  </HorizontalGutter>
);

export default ModerationConfig;
