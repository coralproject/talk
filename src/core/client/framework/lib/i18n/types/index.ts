export { default as FluentShortNumber } from "./FluentShortNumber";
