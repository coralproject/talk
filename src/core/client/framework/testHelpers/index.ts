export {
  default as createRelayEnvironment,
  CreateRelayEnvironmentParams,
} from "./createRelayEnvironment";
export { default as createFluentBundle } from "./createFluentBundle";
export { default as createSinonStub } from "./createSinonStub";
export {
  default as removeFragmentRefs,
  NoFragmentRefs,
} from "./removeFragmentRefs";
export { default as createUUIDGenerator } from "./createUUIDGenerator";
export * from "./denormalize";
export { default as replaceHistoryLocation } from "./replaceHistoryLocation";
