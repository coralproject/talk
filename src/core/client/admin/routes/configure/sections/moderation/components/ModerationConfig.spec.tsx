import { noop } from "lodash";
import React from "react";
import { createRenderer } from "react-test-renderer/shallow";

import { removeFragmentRefs } from "talk-framework/testHelpers";
import { PropTypesOf } from "talk-framework/types";

import ModerationConfig from "./ModerationConfig";

const ModerationConfigN = removeFragmentRefs(ModerationConfig);

it("renders correctly", () => {
  const props: PropTypesOf<typeof ModerationConfigN> = {
    disabled: false,
    settings: {},
    onInitValues: noop,
  };
  const renderer = createRenderer();
  renderer.render(<ModerationConfigN {...props} />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
