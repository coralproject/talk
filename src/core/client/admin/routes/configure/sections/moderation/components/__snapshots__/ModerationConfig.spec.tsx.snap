// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(forwardRef)
  data-testid="configure-moderationContainer"
  size="double"
>
  <Relay(PerspectiveConfigContainer)
    disabled={false}
    onInitValues={[Function]}
    settings={Object {}}
  />
  <Relay(AkismetConfigContainer)
    disabled={false}
    onInitValues={[Function]}
    settings={Object {}}
  />
</ForwardRef(forwardRef)>
`;
